### PR TITLE
[CELEBORN-879] Add `dev/dependencies.sh` for audit dependencies

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

As title

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
Add this to github CI?

**Comment by pan3793** [comment]
@cfmcgrady thanks for working on this.

IMO, the dependencies auditing in Celeborn is a bit complex compared with Spark, since Celeborn is composed of client, master, and worker, all of them have different runtime jars.

But I think we can merge the classpath of master and worker, which simplifies both packaging scripts and dependencies auditing, then the dependencies auditing only needs to take care of client and server dependencies.

- dependencise-client-flink-1.14
- dependencise-client-flink-...
- dependencise-client-spark-2.4
- dependencise-client-spark-...
- dependencise-client-server

**Comment by pan3793** [comment]
And I think we can focus on sbt for this new feature, mvn has some cache issues which are hard to overcome

**Comment by cfmcgrady** [comment]
> Add this to github CI?

I intend to introduce a GitHub Action job once sbt and maven dependencies have been aligned. @AngersZhuuuu

**Comment by cfmcgrady** [comment]
1. Audit the dependencies module by module as suggested.
2. As Maven remains the predominant build tool among users, the dependencies list in this PR is generated using Maven.

**Comment by cfmcgrady** [comment]
the diff between sbt and maven, `-` for maven, `+` for sbt

```diff
diff --git a/dev/deps/dependencies-client-flink-1.14 b/dev/deps/dependencies-client-flink-1.14
index 7e8c870f..bbc28ab9 100644
--- a/dev/deps/dependencies-client-flink-1.14
+++ b/dev/deps/dependencies-client-flink-1.14
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.8.0//lz4-java-1.8.0.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.15//scala-library-2.12.15.jar
 scala-reflect/2.12.15//scala-reflect-2.12.15.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.2-1//zstd-jni-1.5.2-1.jar
diff --git a/dev/deps/dependencies-client-flink-1.15 b/dev/deps/dependencies-client-flink-1.15
index 7e8c870f..bbc28ab9 100644
--- a/dev/deps/dependencies-client-flink-1.15
+++ b/dev/deps/dependencies-client-flink-1.15
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.8.0//lz4-java-1.8.0.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.15//scala-library-2.12.15.jar
 scala-reflect/2.12.15//scala-reflect-2.12.15.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.2-1//zstd-jni-1.5.2-1.jar
diff --git a/dev/deps/dependencies-client-flink-1.17 b/dev/deps/dependencies-client-flink-1.17
index 7e8c870f..bbc28ab9 100644
--- a/dev/deps/dependencies-client-flink-1.17
+++ b/dev/deps/dependencies-client-flink-1.17
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.8.0//lz4-java-1.8.0.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.15//scala-library-2.12.15.jar
 scala-reflect/2.12.15//scala-reflect-2.12.15.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.2-1//zstd-jni-1.5.2-1.jar
diff --git a/dev/deps/dependencies-client-spark-2.4 b/dev/deps/dependencies-client-spark-2.4
index 917a0444..2dd3df6c 100644
--- a/dev/deps/dependencies-client-spark-2.4
+++ b/dev/deps/dependencies-client-spark-2.4
@@ -24,11 +24,9 @@ guava/14.0.1//guava-14.0.1.jar
 hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
-jackson-annotations/2.6.7//jackson-annotations-2.6.7.jar
-jackson-core/2.6.7//jackson-core-2.6.7.jar
-jackson-databind/2.6.7.3//jackson-databind-2.6.7.3.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.4.0//lz4-java-1.4.0.jar
@@ -51,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -76,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.11.12//scala-library-2.11.12.jar
 scala-reflect/2.11.12//scala-reflect-2.11.12.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.4.4-3//zstd-jni-1.4.4-3.jar
diff --git a/dev/deps/dependencies-client-spark-3.0 b/dev/deps/dependencies-client-spark-3.0
index 5005ba52..5ebe3463 100644
--- a/dev/deps/dependencies-client-spark-3.0
+++ b/dev/deps/dependencies-client-spark-3.0
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.7.1//lz4-java-1.7.1.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.10//scala-library-2.12.10.jar
 scala-reflect/2.12.10//scala-reflect-2.12.10.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.4.4-3//zstd-jni-1.4.4-3.jar
diff --git a/dev/deps/dependencies-client-spark-3.1 b/dev/deps/dependencies-client-spark-3.1
index 6e8fd970..772fd3f2 100644
--- a/dev/deps/dependencies-client-spark-3.1
+++ b/dev/deps/dependencies-client-spark-3.1
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.7.1//lz4-java-1.7.1.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.10//scala-library-2.12.10.jar
 scala-reflect/2.12.10//scala-reflect-2.12.10.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.4.8-1//zstd-jni-1.4.8-1.jar
diff --git a/dev/deps/dependencies-client-spark-3.2 b/dev/deps/dependencies-client-spark-3.2
index 6fbebf79..32e829d7 100644
--- a/dev/deps/dependencies-client-spark-3.2
+++ b/dev/deps/dependencies-client-spark-3.2
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.7.1//lz4-java-1.7.1.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.15//scala-library-2.12.15.jar
 scala-reflect/2.12.15//scala-reflect-2.12.15.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.0-4//zstd-jni-1.5.0-4.jar
diff --git a/dev/deps/dependencies-client-spark-3.3 b/dev/deps/dependencies-client-spark-3.3
index 7e8c870f..bbc28ab9 100644
--- a/dev/deps/dependencies-client-spark-3.3
+++ b/dev/deps/dependencies-client-spark-3.3
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.8.0//lz4-java-1.8.0.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -73,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.15//scala-library-2.12.15.jar
 scala-reflect/2.12.15//scala-reflect-2.12.15.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.2-1//zstd-jni-1.5.2-1.jar
diff --git a/dev/deps/dependencies-client-spark-3.4 b/dev/deps/dependencies-client-spark-3.4
index 8c570f49..591177f5 100644
--- a/dev/deps/dependencies-client-spark-3.4
+++ b/dev/deps/dependencies-client-spark-3.4
@@ -25,7 +25,8 @@ hadoop-client-api/3.2.4//hadoop-client-api-3.2.4.jar
 hadoop-client-runtime/3.2.4//hadoop-client-runtime-3.2.4.jar
 htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jcl-over-slf4j/1.7.36//jcl-over-slf4j-1.7.36.jar
-jsr305/1.3.9//jsr305-1.3.9.jar
+jna/4.2.2//jna-4.2.2.jar
+jsr305/3.0.2//jsr305-3.0.2.jar
 jul-to-slf4j/1.7.36//jul-to-slf4j-1.7.36.jar
 leveldbjni-all/1.8//leveldbjni-all-1.8.jar
 lz4-java/1.8.0//lz4-java-1.8.0.jar
@@ -48,6 +49,7 @@ netty-codec-xml/4.1.93.Final//netty-codec-xml-4.1.93.Final.jar
 netty-codec/4.1.93.Final//netty-codec-4.1.93.Final.jar
 netty-common/4.1.93.Final//netty-common-4.1.93.Final.jar
 netty-handler-proxy/4.1.93.Final//netty-handler-proxy-4.1.93.Final.jar
+netty-handler-ssl-ocsp/4.1.93.Final//netty-handler-ssl-ocsp-4.1.93.Final.jar
 netty-handler/4.1.93.Final//netty-handler-4.1.93.Final.jar
 netty-resolver-dns-classes-macos/4.1.93.Final//netty-resolver-dns-classes-macos-4.1.93.Final.jar
 netty-resolver-dns-native-macos/4.1.93.Final/osx-aarch_64/netty-resolver-dns-native-macos-4.1.93.Final-osx-aarch_64.jar
@@ -56,12 +58,8 @@ netty-resolver-dns/4.1.93.Final//netty-resolver-dns-4.1.93.Final.jar
 netty-resolver/4.1.93.Final//netty-resolver-4.1.93.Final.jar
 netty-transport-classes-epoll/4.1.93.Final//netty-transport-classes-epoll-4.1.93.Final.jar
 netty-transport-classes-kqueue/4.1.93.Final//netty-transport-classes-kqueue-4.1.93.Final.jar
-netty-transport-native-epoll/4.1.87.Final/linux-aarch_64/netty-transport-native-epoll-4.1.87.Final-linux-aarch_64.jar
-netty-transport-native-epoll/4.1.87.Final/linux-x86_64/netty-transport-native-epoll-4.1.87.Final-linux-x86_64.jar
 netty-transport-native-epoll/4.1.93.Final/linux-aarch_64/netty-transport-native-epoll-4.1.93.Final-linux-aarch_64.jar
 netty-transport-native-epoll/4.1.93.Final/linux-x86_64/netty-transport-native-epoll-4.1.93.Final-linux-x86_64.jar
-netty-transport-native-kqueue/4.1.87.Final/osx-aarch_64/netty-transport-native-kqueue-4.1.87.Final-osx-aarch_64.jar
-netty-transport-native-kqueue/4.1.87.Final/osx-x86_64/netty-transport-native-kqueue-4.1.87.Final-osx-x86_64.jar
 netty-transport-native-kqueue/4.1.93.Final/osx-aarch_64/netty-transport-native-kqueue-4.1.93.Final-osx-aarch_64.jar
 netty-transport-native-kqueue/4.1.93.Final/osx-x86_64/netty-transport-native-kqueue-4.1.93.Final-osx-x86_64.jar
 netty-transport-native-unix-common/4.1.93.Final//netty-transport-native-unix-common-4.1.93.Final.jar
@@ -77,6 +75,6 @@ ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
 scala-library/2.12.17//scala-library-2.12.17.jar
 scala-reflect/2.12.17//scala-reflect-2.12.17.jar
 shims/0.9.32//shims-0.9.32.jar
-slf4j-api/1.7.36//slf4j-api-1.7.36.jar
+slf4j-api/2.0.7//slf4j-api-2.0.7.jar
 snakeyaml/1.33//snakeyaml-1.33.jar
 zstd-jni/1.5.2-5//zstd-jni-1.5.2-5.jar
```

File: dev/dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    ;;
+  "flink-1.17")
+    MVN_MODULES="client-flink/flink-1.17"
+    SBT_PROJECT="celeborn-client-flink-1_17"
+    ;;
+  *)
+    MVN_MODULES="worker,master"
+    ;;
+esac
+
```
Comment (waitinfuture) [inline dev/dependencies.sh:167]: nit: one blank line

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main.

**Comment by zwangsheng** [comment]
Hi @cfmcgrady, would you like to add how to use this script audit dependencies into `${PROJECT_HOME}/CONTRIBUTING.md`, since we added how to build configurations docs in this.

**Comment by cfmcgrady** [comment]
> Hi @cfmcgrady, would you like to add how to use this script audit dependencies into `${PROJECT_HOME}/CONTRIBUTING.md`, since we added how to build configurations docs in this.

Good catch, Thank you for the reminder, let me fix the ci first.

**Pull request closed without merging**